src/scripts/Game.ts

```diff
@@ -87,9 +87,36 @@ class Game {
         Weather.generateWeather(now);
         RoamingPokemonList.generateIncreasedChanceRoutes(now);
 
+        this.computeOfflineEarnings();
+
         this.gameState = GameConstants.GameState.fighting;
     }
 
+    computeOfflineEarnings() {
+        const now = Date.now();
+        const timeDiffInSeconds = (now - player._previousStartLastSeen) / 1000;
```

Comment:
Just to have a nice round number
```suggestion
        const timeDiffInSeconds = Math.floor((now - player._previousStartLastSeen) / 1000);
```